chaincode/marbles_transfer/marbles_transfer_queries.go

```diff
@@ -106,16 +111,115 @@ func queryMarblePrice(ctx contractapi.TransactionContextInterface, marbleID stri
 	return string(marblePriceJSON), nil
 }
 
-// TODO add a query to get all of an organization's proposed sales
-// Use GetPrivateDataByPartialCompositeKey to find all keys starting with typeMarbleForSale
-// hint: see sample at https://github.com/hyperledger/fabric-samples/blob/master/chaincode/marbles02/go/marbles_chaincode.go#L458
+// QueryMarbleProposedSales returns all of an organization's proposed sales
+func (s *SmartContract) QueryMarbleProposedSales(ctx contractapi.TransactionContextInterface) ([]QueryResult, error) {
+	return queryTxnByType(ctx, typeMarbleForSale)
+}
+
+// QueryMarbleProposedBuys returns all of an organization's proposed buys
+func (s *SmartContract) QueryMarbleProposedBuys(ctx contractapi.TransactionContextInterface) ([]QueryResult, error) {
+	return queryTxnByType(ctx, typeMarbleBid)
+}
+
+func queryTxnByType(ctx contractapi.TransactionContextInterface, txnType string) ([]QueryResult, error) {
+	collection, err := getImplicitCollectionName(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	salesIterator, err := ctx.GetStub().GetPrivateDataByPartialCompositeKey(collection, txnType, []string{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to read from private data collection: %s", err.Error())
+	}
+	defer salesIterator.Close()
+
+	sales := []QueryResult{}
+
+	var i int
+	for i = 0; salesIterator.HasNext(); i++ {
+		resp, err := salesIterator.Next()
+		if err != nil {
+			return nil, err
+		}
+
+		newMarble := new(Marble)
+		err = json.Unmarshal(resp.Value, newMarble)
```

Comment:
The value is a price agreement, not a marble. The price agreement just gets passed around as JSON, not as an object.

Reply:
Should we return the marble ID and price?

Reply:
As discussed, will return marble id, price, and trade id.